content-moderation-evaluation/Evaluating-Amazon-Rekognition-Content-Moderation-Service.ipynb

```diff
@@ -0,0 +1,542 @@
+    "        if test_template\n",
+    "        else \"{{ task.input.labels | to_json | escape }}\",\n",
+    "    )\n",
+    "\n",
+    "    with open(save_fname, \"w\") as f:\n",
+    "        f.write(template)\n",
+    "    if test_template is False:\n",
+    "        print(template)\n",
+    "\n",
+    "\n",
+    "make_template(test_template=True, save_fname=\"instructions.html\")\n",
+    "make_template(test_template=False, save_fname=\"instructions.template\")\n",
+    "s3.upload_file(\"instructions.template\", BUCKET, EXP_NAME + \"/instructions.template\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Define pre-built lambda functions for use in the labeling job\n",
+    "Before we submit the job, we need to define the ARNs for key components of the labeling job: 1) the workteam, 2) the annotation consolidation Lambda function, 3) the pre-labeling task Lambda function, These functions are defined by strings with region names and AWS service account numbers, so we will define a mapping below that will enable you to run this notebook in corresponding AWS region (us-east-1 in our example).\n",
+    "\n",
+    "See the official documentation for the available ARNs:\n",
+    "- Set **VERIFY_USING_PRIVATE_WORKFORCE=False** if you choose to use the [public workfofce](https://docs.aws.amazon.com/sagemaker/latest/dg/sms-workforce-management-public.html) or set **VERIFY_USING_PRIVATE_WORKFORCE=True** if you elect to use a [private workteam](https://docs.aws.amazon.com/sagemaker/latest/dg/sms-workforce-create-private-console.html) and check the corresponding ARN and set variable **private_workteam_arn**.\n",
+    "- [Documentation](https://docs.aws.amazon.com/sagemaker/latest/APIReference/API_HumanTaskConfig.html#SageMaker-Type-HumanTaskConfig-PreHumanTaskLambdaArn) for available pre-human ARNs for other workflows.\n",
+    "- [Documentation](https://docs.aws.amazon.com/sagemaker/latest/APIReference/API_AnnotationConsolidationConfig.html#SageMaker-Type-AnnotationConsolidationConfig-AnnotationConsolidationLambdaArn) for available annotation consolidation ANRs for other workflows."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "private_workteam_arn = \"<ARN OF PRIVATE WORKFORCE>\"\n",
+    "\n",
+    "VERIFY_USING_PRIVATE_WORKFORCE = True\n",
+    "\n",
+    "# Specify ARNs for resources needed to run an image classification job.\n",
+    "ac_arn_map = {\n",
+    "    \"us-east-1\": \"432418664414\",\n",
+    "}\n",
+    "\n",
+    "prehuman_arn = \"arn:aws:lambda:{}:{}:function:PRE-ImageMultiClass\".format(\n",
+    "    region, ac_arn_map[region]\n",
+    ")\n",
+    "\n",
+    "acs_arn = \"arn:aws:lambda:{}:{}:function:ACS-ImageMultiClass\".format(region, ac_arn_map[region])\n",
+    "\n",
+    "workteam_arn = \"arn:aws:sagemaker:{}:394669845002:workteam/public-crowd/default\".format(region)"
```

Comment:
Note: approved for public reference as coming from https://docs.aws.amazon.com/sagemaker/latest/dg/sms-security-workforce-authentication.html